self_test/faceted_surface_vtu_output/tetmesh_faceted_surfaces.h

```diff
@@ -0,0 +1,1256 @@
+ /// assigned. The six faces are enumerated consecutively
+ /// starting from boundary_id_offset (defaults to zero in which
+ /// case the boundaries are enumerated (in one-based fashion) 1,...,6. 
+ CubicTetMeshFacetedSurface(const unsigned& one_based_region_id,
+                            const double& box_half_width, 
+                            const double& box_half_length,
+                            const Vector<double>& offset,
+                            const unsigned& one_based_boundary_id_offset=0)
+  {   
+   build_it(one_based_region_id,
+            box_half_width,
+            box_half_length,
+            offset,
+            one_based_boundary_id_offset);
+  }
+
+
+private:
+
+ /// Build the thing
+ void build_it(const unsigned& one_based_region_id,
+               const double& box_half_width, 
+               const double& box_half_length,
+               const Vector<double>& offset,
+               const unsigned& one_based_boundary_id_offset)
+  {
+   // Make vertices
+   unsigned n_vertex=8;
+   Vertex_pt.resize(n_vertex);
+   Vector<double> box_point(3);
+
+   box_point[0] = offset[0]-box_half_width;
+   box_point[1] = offset[1]-box_half_width;
+   box_point[2] = offset[2]-box_half_length;
+   Vertex_pt[0]=new TetMeshVertex(box_point);
+  
+   box_point[0] = offset[0]-box_half_width;
+   box_point[1] = offset[1]+box_half_width;
+   box_point[2] = offset[2]-box_half_length;
+   Vertex_pt[1]=new TetMeshVertex(box_point);
+
+   box_point[0] = offset[0]-box_half_width;
+   box_point[1] = offset[1]+box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[2]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]-box_half_width;
+   box_point[1] = offset[1]-box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[3]=new TetMeshVertex(box_point);
+
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]-box_half_width;
+   box_point[2] = offset[2]-box_half_length;
+   Vertex_pt[4]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]+box_half_width;
+   box_point[2] = offset[2]-box_half_length;
+   Vertex_pt[5]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]+box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[6]=new TetMeshVertex(box_point);
+   
+   box_point[0] = offset[0]+box_half_width;
+   box_point[1] = offset[1]-box_half_width;
+   box_point[2] = offset[2]+box_half_length;
+   Vertex_pt[7]=new TetMeshVertex(box_point);
+   
+
+   // Make facets
+   unsigned n_facet=6;
+   Facet_pt.resize(n_facet);
+   
+   unsigned n_vertex_on_facet=4;
+   Facet_pt[0]=new TetMeshFacet(n_vertex_on_facet);
+   unsigned one_based_boundary_id=1+one_based_boundary_id_offset;
+   Facet_pt[0]->set_one_based_boundary_id(one_based_boundary_id);   
+   Facet_pt[0]->set_vertex_pt(0,Vertex_pt[0]);
```

Comment:
Please add spaces around the operators and after the comma in `set_vertex_pt(0,Vertex_pt[0]);`

Reply:
...and for the code below